apps/test/DialogButtonsTest.js

```diff
@@ -0,0 +1,35 @@
+window.React = require('react');
+window.ReactDOM = require('react-dom');
+var ReactTestUtils = require('react-addons-test-utils');
+
+var testUtils = require('./util/testUtils');
+var assert = testUtils.assert;
+
+var DialogButtons = require('@cdo/apps/templates/DialogButtons.jsx');
+var msg = require('@cdo/apps/locale');
+
+describe('DialogButtons', function () {
+  it('displays the range of A/B testing strings', function () {
+
+    var isAgainButton = function (childComponent) {
+      return childComponent.id === 'hint-request-button';
+    };
+
+    var testDialogButtonsWithUserId = function (userId, expectedString) {
+      var dialogButtons = <DialogButtons tryAgain={msg.tryAgain()} shouldPromptForHint={true} userId={userId} />;
+      var componentInstance = ReactTestUtils.renderIntoDocument(dialogButtons);
+      var buttons = ReactTestUtils.findAllInRenderedTree(componentInstance, isAgainButton);
+
+      assert.equal(buttons.length, 1);
+      assert.equal(buttons[0].innerText, expectedString);
+    };
+
+    // Undefined or even means "See hint"; odd means "Get a block"
+    testDialogButtonsWithUserId(undefined, msg.hintRequest());
```

Comment:
Though I'm (very, very) fine with things as is (nice tests!)... It seems informative to split this into two tests (signed-in users and non-signed in users). And it seems useful to pull the common setup out of the individual test. Perhaps:

```
describe('DialogButtons', function () {
  var isAgainButton = function (childComponent) {
    return childComponent.id === 'hint-request-button';
  };
  var testDialogButtonsWithUserId = function (userId, expectedString) {
    var dialogButtons = <DialogButtons tryAgain={msg.tryAgain()} shouldPromptForHint={true} userId={userId} />;
    var componentInstance = ReactTestUtils.renderIntoDocument(dialogButtons);
    var buttons = ReactTestUtils.findAllInRenderedTree(componentInstance, isAgainButton);

    assert.equal(buttons.length, 1);
    assert.equal(buttons[0].innerText, expectedString);
  };

  it('displays A/B testing strings for non-signed-in users', function () {
    testDialogButtonsWithUserId(undefined, msg.hintRequest());
  });

  it('displays A/B testing strings for signed-in users', function () {
    testDialogButtonsWithUserId(1, msg.showBlock());
    testDialogButtonsWithUserId(2, msg.hintRequest());
    testDialogButtonsWithUserId(3, msg.showBlock());
    testDialogButtonsWithUserId(4, msg.hintRequest());
  });
});
```

This is me expressing how I would write it, there is no expectation / obligation to follow my lead.